solidity/contracts/token/extensions/HypERC4626OwnerCollateral.sol

```diff
@@ -40,8 +58,11 @@ contract HypERC4626OwnerCollateral is HypERC20Collateral {
     function _transferFromSender(
         uint256 _amount
     ) internal override returns (bytes memory metadata) {
-        metadata = super._transferFromSender(_amount);
+        super._transferFromSender(_amount);
         _depositIntoVault(_amount);
+        rateUpdateNonce++;
```

Comment:
is it not possible to use the message nonce for this?

Reply:
Better to maintain consistency with the Hyp4626Collateral

Reply:
I guess that cant use message nonce because we could skip some rebases?